x/yarpctest/types/stream.go

```diff
@@ -100,6 +102,11 @@ func (s *RecvStreamMsg) applyStream(t testing.TB, c transport.Stream) {
 		}
 		return
 	}
+	if s.WantErr != nil {
+		require.Error(t, err)
+		require.Equal(t, s.WantErr, err)
+		return
```

Comment:
Should we return here or have 

```
if s.WantErr != nil {
  require.Error(t, err)
  require.Equal(t, s.WantErr, err)
} else {
  require.NoError(t, err)
}
```

Reply:
We have to return here to avoid further checks (which are bound to panic)
https://github.com/yarpc/yarpc-go/blob/c94bede5b311086888e4df53adc6697f49590755/x/yarpctest/types/stream.go#L112